### DISCUSSION TITLE
where does buildDog.py run the build CMD ?

### DISCUSSION THREAD

**Discussion by RMartires** [BruceDLong/CodeDog#152] [Q&A] [answered]
Hey guys, 
so i was checking out the current build method in buildDog and i'm a lil confused by one thing.
so when builddog.buid() is called it gets directed to the respective platform build function where it 
(a) creates the files and (b) gets the build string & run string but then it returns. so when does it run the build sting cmd ? 

like shouldn't it have to run the build sting to create the executable file ?

-Thanks

**Comment by BruceDLong** [comment, marked as answer]
Good question!
It uses a python function called subprocess.Popen() which stands for
process open.
It's called on lines 19 and 162. I'm not sure why it's called in two
different places.

The thing about Popen is that it gets back all the information written to
stdout and stderr so it can check for success. Errors go to stderr and
other compile messages go to stdout.

https://docs.python.org/3/library/subprocess.html#popen-constructor

BTW, my normal email address is: ***@***.***

Let me know if that answer is insufficient!

Bruce

On Thu, Jun 10, 2021 at 5:01 PM Rohit Martires ***@***.***>
wrote:

> Hey guys,
> so i was checking out the current build method in buildDog and i'm a lil
> confused by one thing.
> so when builddog.buid() is called it gets directed to the respective
> platform build function where it
> (a) creates the files and (b) gets the build string & run string but then
> it returns. so when does it run the build sting cmd ?
>
> like shouldn't it have to run the build sting to create the executable
> file ?
>
> -Thanks
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BruceDLong/CodeDog/discussions/152>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEZFEUMR75XDMF2I5YBXJ3TSE75NANCNFSM46PRBXHQ>
> .
>


-- 
Give me immortality or give me death!

**Reply by RMartires** [reply]
got it, thanks

**Comment by BruceDLong** [comment]
I think it's hard to figure out how it builds because the function that
does it is called printResults(). We should rename it to something like
buildExecutableAndPrintResults(). Feel free to gut that file and start over.

On Thu, Jun 10, 2021 at 10:51 PM Rohit Martires ***@***.***>
wrote:

> got it, thanks
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/BruceDLong/CodeDog/discussions/152#discussioncomment-856584>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEZFEURVRGRO2KY5EDFLGTTSGI5DANCNFSM46PRBXHQ>
> .
>


-- 
Give me immortality or give me death!
